app/controllers/authenticate_controller.rb

```diff
@@ -163,17 +163,17 @@ def handle_authentication_error(err)
     when Errors::Authentication::RequestBody::MissingRequestParam
       raise BadRequest
 
-    when Errors::Authentication::AuthnOidc::IdTokenExpired
+    when Errors::Authentication::Jwt::TokenExpired
       raise Unauthorized.new(err.message, true)
 
-    when Errors::Authentication::AuthnOidc::ProviderDiscoveryTimeout
+    when Errors::Authentication::OAuth::ProviderDiscoveryTimeout
       raise GatewayTimeout
 
     when Errors::Util::ConcurrencyLimitReachedBeforeCacheInitialization
       raise ServiceUnavailable
 
-    when Errors::Authentication::AuthnOidc::ProviderDiscoveryFailed,
-      Errors::Authentication::AuthnOidc::ProviderFetchCertificateFailed
+    when Errors::Authentication::OAuth::ProviderDiscoveryFailed,
```

Comment:
As spoken about let's think if OAuth is the correct naming b/c not all uses for JWT are Oauth

Reply:
This is not a JWT thing. It will always be an OAuth provider.